### PR TITLE
Set null slug when source is empty

### PR DESCRIPTION
**Pull request by PallMallShow**
In my app, i have a slug generation dynamically and my source field can be blank.
Moreover, I set the unique option to `true`, so, after every insert with a empty field, I had slugs like 

- '-1'
- '-2'
- '-3'
....

I have not done phpunit tests because I prefer to let a professional do it !

### CODE REVIEW

File: src/SluggableTrait.php [hunk 1]
```diff
@@ -80,6 +81,10 @@ protected function generateSlug($source) {
 			$slug = substr($slug, 0, $max_length);
 		}
 
+		if($null_when_empty && empty($source)){
+			$slug = null;
+		}
+
```
Comment (cviebrock) [inline src/SluggableTrait.php]: Looks good, but why not move this check up to line 71.  If the $source is empty, no point in running through all the other code, right?

**Comment by cviebrock** [comment]
Makes sense to me.  In fact, I'd almost suggest that this be the default behaviour.  Can you think of a reason you'd _want_ to have `-1`, `-2`, etc. as your slugs?

(Also see the line comment I made in the diffs for a quick suggestion on improving your MR.)

**Comment by MyoStudio** [comment]
Hi guy,

Yes, you're right, I can move this test to line 71. 
I do this today (after my big deploy :computer: )
Personnaly, I can't say if the default behavior like '-1', '-2' is excepted by developers. It's for this reason I had a parameters, to keep the original behavior but override it in cases.

**Comment by PallMallShow** [comment]
Hi guy,

Yes, you're right, I can move this test to line 71. 
I do this today (after my big deploy :computer: )
Personnaly, I can't say if the default behavior like '-1', '-2' is excepted by developers. It's for this reason I had a parameters, to keep the original behavior but override it in cases.

**Comment by MyoStudio** [comment]
I set slug null in first time before all others tests.
If you want to keep this behavior, no problem.
But if you want to set this the default behavior, tell me, I can set this ;)

**Comment by PallMallShow** [comment]
I set slug null in first time before all others tests.
If you want to keep this behavior, no problem.
But if you want to set this the default behavior, tell me, I can set this ;)

**Comment by cviebrock** [comment]
Yeah, let's make this the default behaviour.  I can't think of a use-case where the source string for a slug would be `null` and yet you'd still want a slug like `-1` to be generated.

Worst case is that user's can overload the `generateSlug()` method in the trait.

Thanks!

**Comment by PallMallShow** [comment]
Hi Colin,

I've removed my super param and set null when the source was empty.
Do you need any change in PHPUnit tests ?

**Comment by cviebrock** [comment]
Merged .. thanks!  I added a unit test to check that it works:

```php
    public function testEmptySourceGeneratesEmptySlug()
    {
        $post = new Post(['title' => 'My Test Post']);
        $post->setSlugConfig([
          'build_from' => 'subtitle'
        ]);

        $post->save();
        $this->assertEquals($post->slug, null);
    }
```

**Pull request merged**